src/magma/vk/commands/CommandBufferArr.cpp

```diff
@@ -0,0 +1,52 @@
+#include "magma/vk/commands/CommandBufferArr.h"
+
+CommandBufferArr::CommandBufferArr(vk::Device device, vk::CommandPool commandPool, uint32_t count) 
+        : _device(device),
+        _commandPool(commandPool) {
+    _commandBuffers.resize(count);
```

Comment:
This resize doesn't make sense.
On the 14th line you make a copy of std::vector.